momepy/shape.py

```diff
@@ -948,41 +948,36 @@ def _angle(a, b, c):
 
             return angle
 
+        def _calc(geom):
+            angles = []
+            points = list(geom.exterior.coords)  # get points of a shape
+            if len(points) < 3:
```

Comment:
It's small optimization, but since we are using `len(points)` 3 times here, let's declare `npoints = len(points)` once and use that here and in lines 956 & 958.